connector/connect/common/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -128,7 +128,7 @@ message WriteOperationV2 {
 
   // (Optional) A provider for the underlying output data source. Spark's default catalog supports
   // "parquet", "json", etc.
-  string provider = 3;
+  optional string provider = 3;
```

Comment:
There is no need for `optional`.

Reply:
Without `optional`, protobuf doesn't provide `hasXxx` method and we can't check it's set/unset.

Reply:
We can omit `optional` only when we can see the default value is the same as unset, e.g., empty list/map and unset. Even for empty list/map, if we can't see they are the same, we should use `optional`. See #39960 for example.

In this case, the default value for string `""` is different from unset.

Reply:
Sorry I actually miss-remember this: the complete  dropping `optional` is still a proposal but not finalized yet. This line is good.